velox/dwio/parquet/writer/Writer.cpp

```diff
@@ -138,49 +142,101 @@ Writer::Writer(
               "writer_node_{}",
               folly::to<std::string>(folly::Random::rand64())))} {}
 
+void Writer::flush() {
+  if (stagingRows_ > 0) {
+    if (!arrowContext_->writer) {
+      auto arrowProperties =
+          ::parquet::ArrowWriterProperties::Builder().build();
+      PARQUET_THROW_NOT_OK(::parquet::arrow::FileWriter::Open(
+          *arrowContext_->schema.get(),
+          arrow::default_memory_pool(),
+          stream_,
+          arrowContext_->properties,
+          arrowProperties,
+          &arrowContext_->writer));
+    }
+
+    auto fields = arrowContext_->schema->fields();
+    std::vector<std::shared_ptr<arrow::ChunkedArray>> chunks;
+    for (int colIdx = 0; colIdx < fields.size(); colIdx++) {
+      auto dataType = fields.at(colIdx)->type();
+      auto chunk =
+          arrow::ChunkedArray::Make(
+              std::move(arrowContext_->stagingChunks.at(colIdx)), dataType)
+              .ValueOrDie();
+      chunks.push_back(chunk);
+    }
+    auto table = arrow::Table::Make(
+        arrowContext_->schema, std::move(chunks), stagingRows_);
+    PARQUET_THROW_NOT_OK(
+        arrowContext_->writer->WriteTable(*table, rowsInRowGroup_));
```

Comment:
Hi, I notice that velox here uses `WriteTable`. Would you mind use `WriteRecordBatch`? This api doesn't flush row-group when called

Reply:
IIUC , the main issue is that, arrow does not support limit the max bytes per row group. This prs cache the RecordBatch and flush as a row group once the size over expected.

Reply:
So you just want a "raw-size" for flushing? Or you need a size after compression?

Reply:
yea, a `raw-size` for flushing is good enough, and this is what this pr does.